src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+                        realArgs=(/minval(sigmamax), maxval(sigmamax)/))
+             call mpas_log_write("damage value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(damage), maxval(damage)/))
+             call mpas_log_write("A value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(meanFlowParamA), maxval(meanFlowParamA)/))
+             call mpas_log_write("alpha value range: Min=$r, Max=$r", &
+                        realArgs=(/minval(alpha), maxval(alpha)/))
+        end if
+        ! temporary message output in the log file
+
+
+         call mpas_deallocate_scratch_field(calvingFrontMaskField, .true.)
+         call mpas_deallocate_scratch_field(meanFlowParamAField, .true.)
+         !call mpas_deallocate_scratch_field(viscField, .true.)
+         call mpas_deallocate_scratch_field(sigmaxxField, .true.)
+         call mpas_deallocate_scratch_field(sigmayyField, .true.)
+         call mpas_deallocate_scratch_field(sigmaminField, .true.)
+         !call mpas_deallocate_scratch_field(sigmamaxField, .true.)
+         call mpas_deallocate_scratch_field(alphaField, .true.)
+         !call mpas_deallocate_scratch_field(epsmaxField, .true.)
+         !call mpas_deallocate_scratch_field(nstarField, .true.)
+         !call mpas_deallocate_scratch_field(szeroField, .true.)
+         !call mpas_deallocate_scratch_field(sourceField, .true.)
+         !call mpas_deallocate_scratch_field(ddamagedtField, .true.)
+         call mpas_deallocate_scratch_field(forceField, .true.)
+         call mpas_deallocate_scratch_field(meanTField, .true.)
+         ! call mpas_deallocate_scratch_field(damageField, .true.)
+
+         block => block % next
+             
+      enddo
+
+   end subroutine calculate_damage 
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!    routine restore_damage
+!
+!> \author Tong Zhang
+!> \date   Oct. 2019
+!> \details restore the damage to the original value before advection if it exceeds some large number
+!-----------------------------------------------------------------------
+
+   subroutine restore_damage(domain, err)
+
+      !-----------------------------------------------------------------
+      ! input variables
+      !-----------------------------------------------------------------
+
+      !-----------------------------------------------------------------
+      ! input/output variables
+      !-----------------------------------------------------------------
+      type (domain_type), intent(inout) :: &
+         domain          !< Input/Output: domain object
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+      integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      ! local variables
```

Comment:
I found a number of variables declared that are not actually used.  Please check the others to make sure there are not any that don't get used.

Reply:
I checked the restore_damage subroutine and deleted all variables that are not used. I'll also keep looking at other places.